src/server.c

```diff
@@ -6245,6 +6251,7 @@ static void sigShutdownHandler(int sig) {
 
     serverLogFromHandler(LL_WARNING, msg);
     server.shutdown_asap = 1;
+    server.last_sig_received = sig;
```

Comment:
we may need to clear that if the shutdown is later canceled, where we do `shutdown_asap = 0;`.
and also on startup (`initServerConfig`, or better yet, `initServer`)